src/Facets/ChainflipFacet.sol

```diff
@@ -0,0 +1,250 @@
+        ILiFi.BridgeData memory _bridgeData,
+        ChainflipData calldata _chainflipData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _chainflipData);
+    }
+
+    /// @notice Performs a swap before bridging via Chainflip
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _chainflipData Data specific to Chainflip
+    function swapAndStartBridgeTokensViaChainflip(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        ChainflipData calldata _chainflipData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _chainflipData);
+    }
+
+    /// Internal Methods ///
+
+    /// @notice Contains the business logic for bridging via Chainflip
+    /// @param _bridgeData The core information needed for bridging, including sending/receiving details
+    /// @param _chainflipData Data specific to Chainflip, including Chainflip token identifiers and parameters
+    /// @dev Handles both EVM and non-EVM destinations, native and ERC20 tokens, and cross-chain messaging
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        ChainflipData calldata _chainflipData
+    ) internal {
+        uint32 dstChain = _getChainflipChainId(_bridgeData.destinationChainId);
+
+        // Handle address encoding based on destination chain type
+        bytes memory encodedDstAddress;
+        if (_bridgeData.receiver == LibAsset.NON_EVM_ADDRESS) {
+            if (_chainflipData.nonEVMReceiver == bytes32(0)) {
+                revert EmptyNonEvmAddress();
+            }
+            encodedDstAddress = abi.encodePacked(
+                _chainflipData.nonEVMReceiver
+            );
+
+            emit BridgeToNonEVMChain(
+                _bridgeData.transactionId,
+                _bridgeData.destinationChainId,
+                _chainflipData.nonEVMReceiver
+            );
+        } else {
+            // For EVM chains, use dstCallReceiver if there's a destination call, otherwise use bridge receiver
+            encodedDstAddress = abi.encodePacked(
+                _bridgeData.hasDestinationCall
+                    ? _chainflipData.dstCallReceiver
```

Comment:
would probably make sense to check that this address has a non-zero value, thats typically something that pops up in the audit as otherwise funds will be lost

wdyt?

Reply:
updated